h2o-algos/src/main/java/hex/glm/ConstrainedGLMUtils.java

```diff
@@ -0,0 +1,861 @@
+        double constrVal = redundantConst._constraints.get(coefName);
+        if (constrVal != 0) {
+          if ("constant".equals(coefName)) {
+            constantVal = constrVal;
+          } else if (trainNames.contains(coefName)) {
+            colInd = trainNames.indexOf(coefName) - coefOffset;
+            if (standardize && colInd >= 0 && !isBetaConstraint) {
+              if (constrVal > 0)
+                sb.append('+');
+              sb.append(constrVal * dinfo._normMul[colInd]);
+            } else {
+              sb.append(constrVal);
+            }
+            sb.append('*');
+            sb.append(coefName);
+          }
+        }
+      }
+      // add constant value here
+      // add constant value to the end
+      if (constantVal != 0) {
+        if (constantVal > 0)
+          sb.append("+");
+        if (isBetaConstraint && colInd >= 0)
+          sb.append(constantVal * dinfo._normMul[colInd]);
+        else
+          sb.append(constantVal);
+      }
+      sb.append(" <= or == 0.");
+      sb.append(" Please remove it from your beta/linear constraints.");
+      return sb.toString();
+    } else {
+      return null;
+    }
+  }
+  
+  public static LinearConstraints getConstraintFromIndex(ComputationState state, Integer constraintIndex) {
+    int constIndexWOffset = constraintIndex;
+    if (state._equalityConstraintsBeta != null) {
+      if (constIndexWOffset < state._equalityConstraintsBeta.length) {
+        return state._equalityConstraintsBeta[constIndexWOffset];
+      } else {
+        constIndexWOffset -= state._equalityConstraintsBeta.length;
+      }
+    }
+    
+    if (state._lessThanEqualToConstraintsBeta != null) {
+      if (constIndexWOffset < state._lessThanEqualToConstraintsBeta.length) {
+        return state._lessThanEqualToConstraintsBeta[constIndexWOffset];
+      } else {
+        constIndexWOffset -= state._lessThanEqualToConstraintsBeta.length;
+      }
+    }
+    
+    if (state._equalityConstraintsLinear != null) {
+      if (constIndexWOffset < state._equalityConstraintsLinear.length) {
+        return state._equalityConstraintsLinear[constIndexWOffset];
+      } else {
+        constIndexWOffset -= state._equalityConstraintsLinear.length;
+      }
+    }
+    
+    if (state._lessThanEqualToConstraints != null && constIndexWOffset < state._lessThanEqualToConstraints.length) {
+      return state._lessThanEqualToConstraints[constIndexWOffset];
+    }
+    return null;
+  }
+
+  /***
+   *
+   * For inequality constraints, this method will calculate the constraint value.  If it is <= 0, the constraint will
+   * be set to be inactive, else, it is active.
+   */
+  public static void setActiveConstraints(LinearConstraints[] lessThanEqualToConstraints,
+                                          double[] beta, List<String> coeffNames) {
+    // evaluate all constraint values
+    stream(lessThanEqualToConstraints).collect(Collectors.toList()).parallelStream().forEach(constraint -> {
```

Comment:
I remember being advised not to use the `parallelStream` and instead use our parallelization framework. I think the reason was that it can confuse our FJ scheduler or something like that.

Reply:
I remove parallelStream stuff.